enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,472 @@
+CRDs.
+
+## Summary
+
+A key goal of Gateway API is to nurture a diverse ecosystem of implementations.
+These implementations, including OpenShift's, all rely on the same Gateway API
+CRDs.  To these ends, OpenShift must manage the life-cycle of these CRDs in a
+way that enables Red Hat to support the product, minimizes friction for the
+end-user, and avoids causing conflicts with other implementations that rely on
+these CRDs.
+
+## Motivation
+
+Gateway API is an evolving API, and its CRDs change over time.  For core
+OpenShift, we must ensure that we have a version of the CRDs that is compatible
+with our product, and we must handle upgrading these CRDs for our own needs.  We
+must also avoid unnecessarily causing friction for Red Hat layered products,
+partner products, and other third-party implementations.
+
+### User Stories
+
+#### Using OpenShift's Gateway API implementation on a new OpenShift cluster
+
+As a cluster-admin, I want to install a new OpenShift 4.19 cluster, and then use
+OpenShift's Gateway API implementation to configure ingress to workload on this
+cluster.
+
+_TBD_
+
+#### Using OpenShift's Gateway API implementation on an upgraded cluster
+
+As a cluster-admin, I want to upgrade my cluster from OpenShift 4.18 (which
+*doesn't* manage the Gateway API CRDs' life-cycle) to OpenShift 4.19 (which
+*does*).
+
+_TBD_
+
+#### Using a third-party Gateway API implementation
+
+As a cluster-admin, I want to install a third-party Gateway API implementation
+on my OpenShift 4.19 cluster, and I want to use the third-party implementation
+without interference from OpenShift.
+
+_TBD_
+
+#### Future OpenShift upgrades
+
+As a cluster-admin, I want to upgrade my cluster from OpenShift 4.19 (which
+manages the Gateway API CRDs' life-cycle) to OpenShift 4.20 (which includes a
+newer version of these CRDs).
+
+_TBD_
+
+### Goals
+
+- Ensure Gateway API CRDs are installed on new and upgraded OpenShift 4.19 clusters.
+- Ensure the installed Gateway API CRDs are compatible with OpenShift's needs.
+- Detect and warn if an incompatible version of the CRDs is installed.
+- Describe the issue of dead fields and potential risks thereof.
+
+### Non-Goals
+
+- Automatically replace incompatible CRDs that some other agent installed.
+- Provide a pre-upgrade check in OpenShift 4.18 for incompatible CRDs.
+- Provide an explicit override for the cluster-admin to take CRD ownership.
+- Implement an admission webhook to block updates to dead fields.
+- Solve CRD life-cycle management in OLM.
+- Solve OLM subscription management.
+- Solve CRD life-cycle management for OSSM or Istio resources.
+
+## Proposal
+
+### Life-cycle management
+
+OpenShift's Ingress Operator monitors for the presence of the Gateway API CRDs.
+
+- If they are absent, it installs the appropriate version.
+- If they are present at the expected version, the operator does nothing.
```

Comment:
If "do nothing" also implies we will use these CRDs I think we'll need to think on that more because we need to manage their lifecycle, and we would need to know who the owner is and have that consent. This however I can add in one of the follow up iterations, so a TODO here should be good:

```suggestion
- If they are present at the expected version: TODO - expand upon what we'll do for preexisting CRDs.
```

Reply:
Are you alluding to transferring ownership with Server-Side Apply?  Now that you mention it, I think the proposal should be rewritten in terms of who owns it rather than whether it is at the expected version.  Should I go ahead and s/at the expected version/with the Ingress Operator as the owner/ now, with a to-do to fill in the SSA-related details?

Reply:
Yes that sounds great :+1:

Reply:
I rewrote in terms of SSA and added a to-do in https://github.com/openshift/enhancements/compare/99aae59bb1ac50f1ef02e07ff3dcc3385216c451..e34441f0d0685989e6417db70dcf049c68f9fc55.